src/components/TodoList/TodoList.tsx

```diff
@@ -0,0 +1,52 @@
+import { useContext } from 'react';
+
+import { TodoContext } from '../../context/TodoContext';
+import { getFilteredTodos } from '../../utils/utils';
+import { FilterContext } from '../../context/FilterContext';
+import { TodoItem } from '../TodoItem';
+
+export const TodoList: React.FC = () => {
+  const { todos, setTodos } = useContext(TodoContext);
+  const { selectedFilter } = useContext(FilterContext);
+
+  const toggleAll = () => {
+    setTodos(currentTodos => {
+      const todosCopy = [...currentTodos];
+
+      return todosCopy.map(todo => ({
```

Comment:
The `map` array method returns a new array, so you don't need to make a copy manually.